Sources/Core/Extensions/Foundation/Data+PovioKit.swift

```diff
@@ -15,27 +15,21 @@ public extension Data {
     map { String(format: "%02x", $0) }.joined()
   }
   
+  /// Returns a UTF-8 encoded string.
+  var utf8: String? {
+    String(data: self, encoding: .utf8)
+  }
+  
+  /// Returns a UTF-16 encoded string.
+  var utf16: String? {
+    String(data: self, encoding: .utf16)
+  }
+  
   /// Decodes data to given `type` using given `decoder`.
   /// ```
   /// let decodedUser = responseData.decode(UserResponse.self, with: JSONDecoder())
   /// ```
   func decode<D: Decodable>(_ type: D.Type, with decoder: JSONDecoder) throws -> D {
-    do {
-      return try decoder.decode(type, from: self)
-    } catch let decodingError as DecodingError {
-      switch decodingError {
-      case .typeMismatch(_, let context),
-           .valueNotFound(_, let context),
-           .keyNotFound(_, let context),
-           .dataCorrupted(let context):
-        Logger.debug("Decoding (failure): \(type.self)", params: ["error": context.debugDescription])
-      @unknown default:
-        Logger.debug("Decoding (failure): \(type.self)", params: ["error": "Unknown decoding error."])
-      }
-      throw decodingError
-    } catch {
-      Logger.debug("Decoding (failure): \(type.self)", params: ["error": error.localizedDescription])
-      throw error
-    }
```

Comment:
Why removed?

Reply:
1. Logger is part of the `Utilities` package, do not want `Core` to depend on the subpackage.
2. Logging should be done on the client side.

Reply:
Don't agree with both points.
1. True, but since the Logger is really the core of the whole package, I would leave it in the `Core` package. 
2. That should be part of the library. Client doesn't and shouldn't know about the implementation details.

Reply:
> That should be part of the library. Client doesn't and shouldn't know about the implementation details.

I disagree. A simple extension function such as this one should be completely pure, i.e. it should not modify the global state.

Reply:
I still tend to disagree. Extension is here for a reason, to "extend" the core functionality. By slashing pretty much all the code, the code left is meaningless, no gain for the developer. We have two options:
1) revert the change
2) remove the method completely

Reply:
The function itself is fine, it extends core functionality to use your lingo. But I would not force upon the user as to how they want to handle (and log the errors). 
If it comes to this I will just remove the method.